### PR TITLE
When you unmute using ctrl-m during push to talk, you now see that reflected in the audio icon

### PR DESCRIPTION
**Pull request by amantley**
When you unmute using ctrl-m during push to talk, you now see that reflected in the audio icon

This pr addresses the following jira case:
https://highfidelity.atlassian.net/browse/BUGZ-76?filter=-1

### Test Plan
1) Go to the Audio settings and enable Push To Talk
2) Close the Audio settings
3) Press and hold T and talk, the other person should be able to hear you.
4) Release T. the other person should not be able to hear you.
5) Press CTRL + M , to unmute the mic. This **does** toggle the little mute display on top of the screen, but after toggling the other person does hear you. 
6) when you press t and release you are back into push to talk

https://highfidelity.manuscript.com/f/cases/22426/

### CODE REVIEW

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4298,6 +4298,10 @@ void Application::keyPressEvent(QKeyEvent* event) {
                 if (isMeta) {
                     auto audioClient = DependencyManager::get<AudioClient>();
                     audioClient->setMuted(!audioClient->isMuted());
+                    auto audioScriptingInterface = reinterpret_cast<scripting::Audio*>(DependencyManager::get<AudioScriptingInterface>().data());
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: Don't use reinterpret_cast do downcast, I know you copied this from commit 18b86d550.  I would prefer it was changed to use the [qSharedPointerDynamicCast](https://doc.qt.io/qt-5/qsharedpointer.html#qSharedPointerDynamicCast)() function.  Using dynamic_cast<> is also acceptable.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19146/HighFidelity-Beta-PR15493-4b7b416.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19146/HighFidelity-Quest-Beta-PR15493-4b7b416.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19146/HighFidelity-Beta-PR15493-4b7b416.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19146/HighFidelity-Beta-Interface-PR15493-4b7b416.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19146/HighFidelity-Beta-PR15493-4b7b416.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19146/HighFidelity-Beta-Interface-PR15493-4b7b416.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19161/HighFidelity-Beta-PR15493-a8b54cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19161/HighFidelity-Beta-Interface-PR15493-a8b54cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19161/HighFidelity-Beta-PR15493-a8b54cb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19161/HighFidelity-Beta-Interface-PR15493-a8b54cb.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19161/HighFidelity-Beta-PR15493-a8b54cb.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19161/HighFidelity-Quest-Beta-PR15493-a8b54cb.apk)

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -1634,7 +1634,7 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
     connect(userInputMapper.data(), &UserInputMapper::actionEvent, [this](int action, float state) {
         using namespace controller;
         auto tabletScriptingInterface = DependencyManager::get<TabletScriptingInterface>();
-        auto audioScriptingInterface = reinterpret_cast<scripting::Audio*>(DependencyManager::get<AudioScriptingInterface>().data());
+        auto audioScriptingInterface = dynamic_cast<scripting::Audio*>(DependencyManager::get<AudioScriptingInterface>().data());
```
Comment (hyperlogic) [inline interface/src/Application.cpp]: You need to verify that the pointer is valid after you dynamic cast

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19185/HighFidelity-Beta-PR15493-1f004f3.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19185/HighFidelity-Beta-Interface-PR15493-1f004f3.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19185/HighFidelity-Beta-PR15493-1f004f3.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19185/HighFidelity-Beta-Interface-PR15493-1f004f3.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19185/HighFidelity-Beta-PR15493-1f004f3.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19185/HighFidelity-Quest-Beta-PR15493-1f004f3.apk)

**Pull request merged**